### PR TITLE
Properly catching exit code and emitting error (fixes #10)

### LINKED ISSUE #10: Exit code not being captured
```javascript
let sshExec = require('ssh-exec')

sshExec('exit 1', 'user@example', (err, stdout, stderr) => {
  if (err) {
    console.log('error found')
    console.log(err)
  } else {
    console.log('error not found')
  }
})
```

Outputs `error not found`.

### PR DESCRIPTION
**Pull request by andyfleming**

### CODE REVIEW

**Comment by andyfleming** [comment]
Have you had a chance to look at this @mafintosh?

File: index.js [hunk 1]
```diff
@@ -77,12 +77,13 @@ var exec = function (cmd, opts, cb) {
...
       })
 
-      stdio.on('close', function () {
-        client.end()
-      })
-
       stdio.on('exit', function (code) {
-        stream.emit('exit', code)
+        if (code !== 0) {
+          stream.emit('error', code)
```
Comment (mafintosh) [inline index.js]: this should emit an actual error, `var err = new Error('Non-zero exit code: ' + code)` and `err.code = code`.

File: index.js [hunk 2]
```diff
@@ -128,8 +129,11 @@ var oncallback = function (stream, cb) {
     stdout += data
   })
 
-  stream.on('error', cb)
-  stream.on('end', function () {
+  stream.on('error', function(code) {
+    cb({code: code}, stdout, stderr)
+  })
+
+  stream.on('exit', function () {
```
Comment (mafintosh) [inline index.js]: should still be on end

**Comment by mafintosh** [comment]
@andyfleming hi. added few comments

**Comment by andyfleming** [comment]
Thanks @mafintosh! I'll took a brief look, but I'll review it in a bit more detail. It could just be documentation that we need to add.

Additional Notes:
* I'm using the callback API.
* You can see the specific use case here: [run-remote-shell-command.js#L66-L118](https://github.com/space-race/mc-ext-basics/blob/199763c901112b110949764f89be5132e219da64/src/stages/run-remote-shell-command.js#L66-L118). (that is using my fork with the changes in this PR)

**Comment by mtwalsh** [comment]
Hi,

I gave this a try in a Meteor app I'm developing and I had more success with the 1.4.0 version.  There were occasions when the error didn't seem to be returned and I'm having no such issues with the 1.4.0 version.

I did find it a bit tricky to extract the error message that was returned (1.4.0) but using toString() has given the output I needed although that's probably rather brittle.

**Comment by andyfleming** [comment]
Fair enough, @mtwalsh. I may not know enough about the underlying API to really fix #10 the proper way.

**Comment by andyfleming** [comment]
Ok. I've made the changes you suggested, @mafintosh, and from my local testing, it still works for me as expected.

File: index.js [hunk 3]
```diff
@@ -128,7 +131,10 @@ var oncallback = function (stream, cb) {
     stdout += data
   })
 
-  stream.on('error', cb)
+  stream.on('error', function(code) {
+    cb({code: code}, stdout, stderr)
```
Comment (mafintosh) [inline index.js]: this is just

```
stream.on('error', function (err) {
  cb(err, stdout, stderr)
})
```

Comment (andyfleming) [inline index.js]: Good catch, @mafintosh. Thanks!

**Comment by mafintosh** [comment]
Great. Released in 2.0.0

**Comment by andyfleming** [comment]
Awesome. Thanks! :beers:

**Pull request merged**